TACC/Core-Styles
# refactor: no padding-left for certain form things

## Overview

Avoid complex specificity war caused by elements CSS trying to avoid padding-left on lists in complex forms.*

<sup>\* Because complex form styles are via component or trump, which override element styles.</sup>

## Status

- [ ] fix https://github.com/TACC/Core-Styles/pull/276#issuecomment-1852937933

## Related

- resolves [#275 (comment)](https://github.com/TACC/Core-Styles/pull/275/files#r1409587968)

## Changes

1. (deprecate) **no** supporting `<ul>` in lieue of menu in a c-form field
2. (hack) **let** django-cms-forms override core-styles.cms (and explain why)
3. (cleanup) **no** avoiding padding on form list cuz form already unsets it

## Testing

0. Open all forms in demo.
1. Ensure the following do not have bullets nor padding-left:
    - form errors
    - field errors
    - radio button lists
    - checkbox lists

## UI

<details><summary>c-form</summary>

| c-form--default | c-form--portal | c-form--cms |
| - | - | - |
| ![c-form--default](https://github.com/TACC/Core-Styles/assets/62723358/28a319e8-1be4-4e38-877e-03479515864f) | ![c-form--portal](https://github.com/TACC/Core-Styles/assets/62723358/60d6ac0e-12d2-488f-9fdd-4d50636840c5) | ![c-form--cms](https://github.com/TACC/Core-Styles/assets/62723358/4cfa430d-1762-4255-9cfd-90dd3589472f) |

</details>
<details><summary>django-cms-forms</summary>

![django-cms-forms](https://github.com/TACC/Core-Styles/assets/62723358/730f6da7-1a5b-4708-929d-32812fcc6438)

</details>
<details><summary>s-form</summary>

| s-form--default | s-form--portal | s-form--cms |
| - | - | - |
| ![s-form--default](https://github.com/TACC/Core-Styles/assets/62723358/650f7e17-e440-447b-b1ba-2b2623498fb2) | ![s-form--portal](https://github.com/TACC/Core-Styles/assets/62723358/e95d8701-16dd-4d92-9c47-303a7ff5a435) | ![s-form--cms](https://github.com/TACC/Core-Styles/assets/62723358/2f505479-8296-424d-95b4-2c2111b93a13) |

</details>

## Notes

This seems backwards-compatible, because any client using a `c-form` has not been using `<ul>`.
